### PR TITLE
refactor: changed the CLI to have different subcommands e.g `resolve` and `install`

### PR DESCRIPTION
**Pull request by tdejager**
This adds a more flexible way to add commands to the binary. I want to update the GIF once this lands so that you do not see the ugly branch name in the gif.

### CODE REVIEW

File: crates/rattler_installs_packages/src/resolve/solve_options.rs [hunk 1]
```diff
@@ -237,3 +237,5 @@ impl Default for ResolveOptions {
         }
     }
 }
+
+// ResolveOptions.with_max_concurrent_tasks(30).with_locked_packages
```
Comment (nichmor) [inline crates/rattler_installs_packages/src/resolve/solve_options.rs:241]: I think it is accidentally committed

**Review by nichmor — APPROVED** [review]
I really love it!

File: crates/rattler_installs_packages/src/resolve/solve_options.rs (hunk 1 above)
Comment (tdejager) [reply crates/rattler_installs_packages/src/resolve/solve_options.rs:241]: Haha :D

File: crates/rip_bin/src/main.rs [hunk 2]
```diff
@@ -344,13 +91,23 @@ async fn main() {
...
 /// Constructs a default [`EnvFilter`] that is used when the user did not specify a custom RUST_LOG.
-pub fn get_default_env_filter(verbose: bool) -> EnvFilter {
-    let mut result = EnvFilter::new("rip=info")
-        .add_directive(Directive::from_str("rattler_installs_packages=info").unwrap());
-
-    if verbose {
-        result = result.add_directive(Directive::from_str("resolvo=info").unwrap());
-    }
+pub fn get_default_env_filter(verbose: clap_verbosity_flag::Verbosity) -> EnvFilter {
+    // Always log info for rattler_installs_packages
```
Comment (nichmor) [inline crates/rip_bin/src/main.rs:95]: what do you mean by always log info for rattler_installs_packages?
I see that we are returning it always for rip

Comment (tdejager) [reply crates/rip_bin/src/main.rs:95]: Yeah its the same new but `rip` is kind of the alias, we wanted to use that but it was taken on crates.io, so the longer name is the official crate name.

Comment (nichmor) [reply crates/rip_bin/src/main.rs:95]: got it - thank you for explaining it

**Pull request merged**